### DISCUSSION TITLE
Automatic builds of CatBoost using GitHub actions

### DISCUSSION THREAD

**Discussion by andrey-khropov** [catboost/catboost#2708] [Announcements]
Hello everyone!

We've set up automatic builds of CatBoost using GitHub actions and now you can obtain all artifacts (CLI binaries, libraries, R packages, python wheels) built from each commit on GitHub.

In order to do this, you have to go to the 'Actions' tab for the CatBoost repository: https://github.com/catboost/catboost/actions, click on a workflow run for .github/workflows/check.yaml for the needed commit and you can find all artifacts in the 'Artifacts' section.

You have to be signed into GitHub to download artifacts (https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts)

Each artifact is a zip archive (GitHub mandates this), you can find a file or files in the expected format after unpacking it.

Artifacts are stored for three days after the commit check.

CatBoost automic build using GitHub actions is initiated by the presence of specified workflows in .github/workflows (https://github.com/catboost/catboost/tree/master/.github/workflows) and uses free GitHub-hosted runners (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories), so if you clone the main CatBoost repository to your own public repository on GitHub, autobuild and check will also automatically run for your commits there (but you will have to manually run the workflow https://github.com/<your_name>/catboost/actions/workflows/add_mac_os_extra_env_to_cache.yaml first). If you want to disable this behaviour you can delete https://github.com/<your_name>/catboost/blob/master/.github/workflows/check.yaml or modify workflow run conditions in it.

**Comment by mrtclb2301-cyber** [comment]
I caanot load into R STUDİO

**Reply by andrey-khropov** [reply]
> I caanot load into R STUDİO

I don't understand what do you mean by that.
